### PR TITLE
V4.0.1 Phase 1 - Adding missing events

### PR DESCRIPTION
**Pull request by mikiquantum**
This PR covers updating or adding the most common events, we will keep adding more in the upcoming weeks.

### CODE REVIEW

File: teste2e/main_test.go [hunk 1]
```diff
@@ -36,6 +36,7 @@ func TestMain(m *testing.M) {
 	localApi, err := gsrpc.NewSubstrateAPI(config.Default().RPCURL)
 	subscriptionsAPI = localApi
 	assert.NoError(&testing.T{}, err)
+	//m.Run()
```
Comment (NunoAlexandre) [inline teste2e/main_test.go:39]: Do we want to remove this?

File: teste2e/author_submit_and_watch_extrinsic_test.go [hunk 2]
```diff
@@ -86,6 +86,7 @@ func TestAuthor_SubmitAndWatchExtrinsic(t *testing.T) {
 
 		sub, err = api.RPC.Author.SubmitAndWatchExtrinsic(ext)
 		if err != nil {
+			nonce++
```
Comment (NunoAlexandre) [inline teste2e/author_submit_and_watch_extrinsic_test.go:89]: Why is this required now / how did it work before without this increment?

File: types/event_record.go [hunk 3]
```diff
@@ -57,161 +57,230 @@ func (e *EventRecordsRaw) Decode(decoder scale.Decoder) error {
...
-	Indices_IndexFrozen                []EventIndicesIndexFrozen                //nolint:stylecheck,golint
-	Offences_Offence                   []EventOffencesOffence                   //nolint:stylecheck,golint
-	Session_NewSession                 []EventSessionNewSession                 //nolint:stylecheck,golint
-	Staking_EraPayout                  []EventStakingEraPayout                  //nolint:stylecheck,golint
-	Staking_Reward                     []EventStakingReward                     //nolint:stylecheck,golint
-	Staking_Slash                      []EventStakingSlash                      //nolint:stylecheck,golint
-	Staking_OldSlashingReportDiscarded []EventStakingOldSlashingReportDiscarded //nolint:stylecheck,golint
-	Staking_StakingElection            []EventStakingStakingElection            //nolint:stylecheck,golint
-	Staking_SolutionStored             []EventStakingSolutionStored             //nolint:stylecheck,golint
+	// Add Auctions
```
Comment (NunoAlexandre) [inline types/event_record.go:64]:
```suggestion
	// TODO: Add Auctions
```

File: types/event_record.go [hunk 4]
```diff
@@ -57,161 +57,230 @@ func (e *EventRecordsRaw) Decode(decoder scale.Decoder) error {
...
+	Balances_Deposit             []EventBalancesDeposit            //nolint:stylecheck,golint
+	Balances_DustLost            []EventBalancesDustLost           //nolint:stylecheck,golint
+	Balances_Endowed             []EventBalancesEndowed            //nolint:stylecheck,golint
+	Balances_Reserved            []EventBalancesReserved           //nolint:stylecheck,golint
+	Balances_ReservedRepatriated []EventBalancesReserveRepatriated //nolint:stylecheck,golint
+	Balances_Slashed             []EventBalancesSlashed            //nolint:stylecheck,golint
+	Balances_Transfer            []EventBalancesTransfer           //nolint:stylecheck,golint
+	Balances_Unreserved          []EventBalancesUnreserved         //nolint:stylecheck,golint
+	Balances_Withdraw            []EventBalancesWithdraw           //nolint:stylecheck,golint
+	// Finish Bounties
```
Comment (NunoAlexandre) [inline types/event_record.go:80]:
```suggestion
	// TODO: Finish Bounties
```

File: types/event_record.go [hunk 5]
```diff
@@ -57,161 +57,230 @@ func (e *EventRecordsRaw) Decode(decoder scale.Decoder) error {
...
+	CollatorSelection_NewCandidacyBond     []EventCollatorSelectionNewCandidacyBond     //nolint:stylecheck,golint
+	CollatorSelection_CandidateAdded       []EventCollatorSelectionCandidateAdded       //nolint:stylecheck,golint
+	CollatorSelection_CandidateRemoved     []EventCollatorSelectionCandidateRemoved     //nolint:stylecheck,golint
+	Contracts_CodeRemoved                  []EventContractsCodeRemoved                  //nolint:stylecheck,golint
+	Contracts_CodeStored                   []EventContractsCodeStored                   //nolint:stylecheck,golint
+	Contracts_ContractCodeUpdated          []EventContractsContractCodeUpdated          //nolint:stylecheck,golint
+	Contracts_ContractEmitted              []EventContractsContractEmitted              //nolint:stylecheck,golint
+	Contracts_Instantiated                 []EventContractsInstantiated                 //nolint:stylecheck,golint
+	Contracts_Terminated                   []EventContractsTerminated                   //nolint:stylecheck,golint
+	// Add Conviction voting
```
Comment (NunoAlexandre) [inline types/event_record.go:101]:
```suggestion
	// TODO: Add Conviction voting
```

File: types/event_record.go [hunk 6]
```diff
@@ -57,161 +57,230 @@ func (e *EventRecordsRaw) Decode(decoder scale.Decoder) error {
...
+	Council_Approved       []EventCollectiveApproved       //nolint:stylecheck,golint
+	Council_Closed         []EventCollectiveClosed         //nolint:stylecheck,golint
+	Council_Disapproved    []EventCollectiveDisapproved    //nolint:stylecheck,golint
+	Council_Executed       []EventCollectiveExecuted       //nolint:stylecheck,golint
+	Council_MemberExecuted []EventCollectiveMemberExecuted //nolint:stylecheck,golint
+	Council_Proposed       []EventCollectiveProposed       //nolint:stylecheck,golint
+	Council_Voted          []EventCollectiveVoted          //nolint:stylecheck,golint
+	// Add Crowdloan
```
Comment (NunoAlexandre) [inline types/event_record.go:109]:
```suggestion
	// TODO: Add Crowdloan
```

**Review by NunoAlexandre — APPROVED** [review]
Looks good to me, just left a few questions 👍

File: teste2e/author_submit_and_watch_extrinsic_test.go (hunk 2 above)
Comment (mikiquantum) [reply teste2e/author_submit_and_watch_extrinsic_test.go:89]: It was all a matter of timing, CI did fail from time to time, I am hoping that this helps.

File: teste2e/main_test.go (hunk 1 above)
Comment (mikiquantum) [reply teste2e/main_test.go:39]: yeah

**Review by NunoAlexandre — APPROVED** [review]

**Comment by cdamian** [comment]
Closing this in favor of - https://github.com/centrifuge/go-substrate-rpc-client/pull/258
